oggm/tests/test_models.py

```diff
@@ -735,52 +827,68 @@ def test_uncertain_mb(self, hef_gdir):
 
         gdir = hef_gdir
 
-        ref_mod = massbalance.ConstantMassBalance(gdir, bias=0)
-        mb_mod = massbalance.UncertainMassBalance(ref_mod)
+        ref_mod = massbalance.ConstantMassBalance(gdir, bias=0,
+                                                  check_calib_params=False)
 
+        # only change bias: this works as before
+        mb_mod = massbalance.UncertainMassBalance(ref_mod,
+                                                  rdn_temp_bias_sigma=0,
+                                                  rdn_prcp_bias_sigma=0,
+                                                  rdn_bias_sigma=100)
         yrs = np.arange(100)
         h, w = gdir.get_inversion_flowline_hw()
         ref_mb = ref_mod.get_specific_mb(h, w, year=yrs)
         unc_mb = mb_mod.get_specific_mb(h, w, year=yrs)
         check_mb = ref_mod.get_specific_mb(h, w, year=yrs)
-        unc2_mb = mb_mod.get_specific_mb(h, w, year=yrs)
-
         assert_allclose(ref_mb, check_mb)
-        assert_allclose(unc_mb, unc2_mb)
         assert np.std(unc_mb) > 50
 
-        mb_mod = massbalance.UncertainMassBalance(ref_mod,
-                                                  rdn_temp_bias_sigma=0.1,
-                                                  rdn_prcp_bias_sigma=0,
-                                                  rdn_bias_sigma=0)
+        mb_mod = massbalance.UncertainMassBalance(ref_mod)
         ref_mb = ref_mod.get_specific_mb(h, w, year=yrs)
+        temp_1 = ref_mod.mbmod.temp.copy()
         unc_mb = mb_mod.get_specific_mb(h, w, year=yrs)
+        temp_2 = ref_mod.mbmod.temp.copy()
         check_mb = ref_mod.get_specific_mb(h, w, year=yrs)
-        assert_allclose(ref_mb, check_mb)
+        temp_3 = ref_mod.mbmod.temp.copy()
+        unc2_mb = mb_mod.get_specific_mb(h, w, year=yrs)
+        # problem: UncertainMassBalance does something to ref_mod,
+        # it slightly changes the precipitation and temperature time series
+        # maybe there is a problem because of rounding issues???
+        # because I always update the time/prcp time series immediately...
+        # this might be the reason that the tests fail, but actually the tests
+        # fail more with larger _sigma, (see below)
+        # so there must be something in ref_mb that changes ref_mod stronger
+        # with a larger _sigma, the temp_bias and prcp_fac stay however the same
+        assert_allclose(temp_1, temp_2, rtol=1e-5)
+        assert_allclose(temp_1, temp_3, rtol=1e-5)
```

Comment:
I am totally confused what is happening here, @fmaussion  , do you have an idea what the problem is, it can't be only because of rounding issues?!

Reply:
yes rounding issues very likely. Whether it's acceptable or not depends on the test and the change in accuracy I guess, will have a look